pkgs/development/cuda-modules/setup-hooks/extension.nix

```diff
@@ -1,74 +1,74 @@
-final: _: {
+# NOTE(@connorbaker): None of the functions in this attribute set should need to access _final or _prev.
+# As such, they are prefixed with an underscore -- everything should be doable with the spliced package sets
+# provided to each function in the attribute set by `final.callPackage`.
+let
+  createSetupHooks =
+    setupHooksAttrs: final: prev:
+    let
+      # It is imperative that we use `final.callPackage` to create these setup hooks, as it allows us access to the spliced
+      # package sets.
+      inherit (final) callPackage;
+
+      # NOTE(@connorbaker): We MUST use `lib` from `prev` because the attribute names CAN NOT depend on `final`.
+      inherit (prev.lib.attrsets) mapAttrs;
+    in
+    mapAttrs (_: value: callPackage value { }) setupHooksAttrs;
+in
+createSetupHooks {
```

Comment:
Not sure I get this. Why do we need `mapAttrs`?

Reply:
`mapAttrs` is used here to apply `callPackage` to the value of each attribute in the attribute set -- it saves us (well... me) from having to wrap the value of each attribute in a call to `callPackage ... {}` (which would increase indentation when using nixfmt).

Reply:
Sure, I can do that!